### ISSUE TITLE
Peer disappear

### ISSUE THREAD

**Issue by ursoft** [ursoft/zwift-offline#1]
https://github.com/zoffline/zwift-offline/issues/157#issuecomment-1041905848
First time I saw this 'heisenbug' between 1/dec/2021 and 7/jan/2022 (after https://github.com/zoffline/zwift-offline/commit/c9483933751c70b4443766eed95e2c2c4d880b97 and before 5f23720c04a9aaf01fb13725b3638cd34aea651b when heading becomes 32 bit).
I reproduced it with my fork about 10 times. Tried to reproduce with original Zwift servers with no luck.

**Comment by oldnapalm** [comment]
When we were investigating this issue (it was reported by an user from the Discord server in May 2021) a place where it was easy to reproduce was here, in the beginning of the "Road to Sky" route

![unknown](https://user-images.githubusercontent.com/38410858/154480305-0f189397-c628-44e3-b90b-e2a9215ad273.png)

If you clicked to watch the remote player you could see them riding in circles close to the green gate (hundreds of meters behind)

![2021-05-13_09_08_14-](https://user-images.githubusercontent.com/38410858/154480581-635ce387-4dac-42a3-af95-9dcc249d9825.png)

Apparently it was fixed (probably just minimized) by https://github.com/zoffline/zwift-offline/commit/c9483933751c70b4443766eed95e2c2c4d880b97

Hopefully https://github.com/ursoft/zwift-offline/commit/4affea6b9cc88d8a2cf2c44c8ab50fba9fd761ce will fix it for good.

**Comment by ursoft** [comment]
Thanks for reproduction conditions, we'll try them

**Comment by ursoft** [comment]
Small test with "Road to Sky" and new code: no issues. 
Starting long series of different tests.

**Comment by ursoft** [comment]
New test results:
Previously, I saw this bug only on weak i7 virtual environment. But today I saw it exclusively on fast machine (and micro #2 too, maybe they are linked together). I have [dumped](https://drive.google.com/file/d/1mGwogNxpVxA1phDxH6cN3CO5d2_wDFB_/view?usp=sharing) UDP/TCP 3xxx (look last two hours: on Y.P. screen his peer Anton P. disappeared about 10 times and after ~100sec pause appeared again at normal place; on A.P. screen both avatars were at normal places). What did I change to move bug from slow PC to fast notebook? I just change resolution on slow machine from 1024p to 720p (same video card) and on fast machine from 2k@144Hz display (Nvidia RTX2060mobile) to 8k@30Hz display (Intel UHD 630, really rendered in maximized window at 4096*2160 resolution via USB-C -> HDMI convertor).
